### PR TITLE
Feature/hot 3

### PR DESCRIPTION
**Pull request by Vladnickgo**

### CODE REVIEW

File: src/main/webapp/WEB-INF/web.xml [hunk 1]
```diff
@@ -7,4 +7,20 @@
...
+    </error-page>
+<!--    <error-page>-->
+<!--        <error-code>500</error-code>-->
+<!--        <location>/error</location>-->
+<!--    </error-page>-->
+    <error-page>
+        <exception-type>com.vladnickgofj.hotel.service.exception.InvalidEmailException</exception-type>
+<!--        <location>/errorEmailPage.jsp</location>-->
+        <location>/error</location>
+
```
Comment (1SpringIsComing1) [inline src/main/webapp/WEB-INF/web.xml:23]: why so much commented lines

File: src/main/java/com/vladnickgofj/hotel/controller/ExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        resp.sendRedirect(PagesConstant.PAGE_NOT_FOUND);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+//        String email = req.getParameter("email");
+//        System.out.println("email "+email);
+//        String code = (String)req.getParameter("javax.servlet.error.status_code");
+//        System.out.println("code: "+code);
+        System.out.println(")))))))))))))");
```
[the remark this answers is not in the document]
Comment (Vladnickgo) [reply src/main/java/com/vladnickgofj/hotel/controller/ExceptionHandler.java:39]: It's already removed

File: src/main/java/com/vladnickgofj/hotel/controller/ExceptionHandler.java [hunk 3]
```diff
@@ -28,42 +32,130 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws S
...
+            }
+
+            if ("AuthorisationFailException".equals(exceptionName)) {
+                req.getSession().invalidate();
+                req.getSession().setAttribute("loginPageEmail", email);
+                req.getSession().setAttribute("authFailed", "true");
+                resp.sendRedirect(PagesConstant.LOGIN_PAGE);
+            }
+
+            if ("RegisterPageEmailErrorException".equals(exceptionName)) {
```
Comment (MaxWel2018) [inline src/main/java/com/vladnickgofj/hotel/controller/ExceptionHandler.java:92]: ***ErorrException not good name for Custom Exception

File: src/main/java/com/vladnickgofj/hotel/controller/ExceptionHandler.java [hunk 4]
```diff
@@ -28,42 +32,130 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws S
...
+                req.getSession().invalidate();
+                req.getSession().setAttribute("firstName", firstName);
+                req.getSession().setAttribute("lastName", lastName);
+                req.getSession().setAttribute("email", email);
+                req.getSession().setAttribute("notValidEmail", "true");
+                LOGGER.info("Not valid Email");
+                resp.sendRedirect(PagesConstant.REGISTRATION_PAGE);
+            }
+            if ("RegisterPageFirstNameErrorException".equals(exceptionName)) {
```
Comment (MaxWel2018) [inline src/main/java/com/vladnickgofj/hotel/controller/ExceptionHandler.java:101]: I don't know why u created exception for inCorrect FirstName

File: src/main/java/com/vladnickgofj/hotel/controller/ExceptionHandler.java [hunk 5]
```diff
@@ -28,42 +32,130 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws S
...
+                req.getSession().invalidate();
+                req.getSession().setAttribute("firstName", firstName);
+                req.getSession().setAttribute("lastName", lastName);
+                req.getSession().setAttribute("email", email);
+                req.getSession().setAttribute("notValidFirstName", "true");
+                LOGGER.info("Not valid Name");
+                resp.sendRedirect(PagesConstant.REGISTRATION_PAGE);
+            }
+            if ("RegisterPageLastNameErrorException".equals(exceptionName)) {
```
Comment (MaxWel2018) [inline src/main/java/com/vladnickgofj/hotel/controller/ExceptionHandler.java:110]: Same for Last name

File: src/main/java/com/vladnickgofj/hotel/controller/ExceptionHandler.java [hunk 6]
```diff
@@ -28,42 +32,130 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws S
...
+                req.getSession().invalidate();
+                req.getSession().setAttribute("firstName", firstName);
+                req.getSession().setAttribute("lastName", lastName);
+                req.getSession().setAttribute("email", email);
+                req.getSession().setAttribute("notValidLastName", "true");
+                LOGGER.info("Not valid lastName");
+                resp.sendRedirect(PagesConstant.REGISTRATION_PAGE);
+            }
+            if ("RegisterPagePasswordErrorException".equals(exceptionName)) {
```
Comment (MaxWel2018) [inline src/main/java/com/vladnickgofj/hotel/controller/ExceptionHandler.java:119]: I don't know why u use RegisterPage in the start of exception's name

File: src/main/java/com/vladnickgofj/hotel/controller/command/home/LoginCommand.java [hunk 7]
```diff
@@ -24,24 +25,23 @@ public LoginCommand(UserService userService) {
...
     public String execute(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        // TODO: 4/7/2022  some logic
         HttpSession session = request.getSession();
         String email = request.getParameter("email");
         String pass = request.getParameter("password");
 
         UserDto byEmail = userService.findByEmail(email);
         if (emailValidation(email)) {
-            request.setAttribute("message", "Email is not correct");
+            throw new LoginPageEmailErrorException("Email not correct");
```
Comment (MaxWel2018) [inline src/main/java/com/vladnickgofj/hotel/controller/command/home/LoginCommand.java:34]: should be default exception IlligalArgumentException

File: src/main/java/com/vladnickgofj/hotel/controller/command/home/LoginCommand.java [hunk 8]
```diff
@@ -24,24 +25,23 @@ public LoginCommand(UserService userService) {
...
         }
         if (email.equals(byEmail.getEmail()) && pass.equals(byEmail.getPassword())) {
             User user = new UserMapper().mapDtoToEntity(byEmail);
             user.setPassword(StringUtils.EMPTY);
             session.setAttribute("user", user);
         } else {
-            response.sendError(HttpStatusCodes.NOT_CORRECT_PASSWORD);
+            request.setAttribute("statusCode",HttpStatusCodes.NOT_CORRECT_PASSWORD);
```
Comment (MaxWel2018) [inline src/main/java/com/vladnickgofj/hotel/controller/command/home/LoginCommand.java:41]: why u use status code here

File: src/main/java/com/vladnickgofj/hotel/validator/UserValidator.java [hunk 9]
```diff
@@ -13,26 +19,34 @@
...
-        validateByParam(UserDto::getPassword, PASSWORD_PATTERN, PASSWORD_ERROR_MESSAGE, entity);
-        validateByParam(UserDto::getConfirmationPassword, PASSWORD_PATTERN, CONFIRMATION_PASSWORD_ERROR_MESSAGE, entity);
+        validateByParam(UserDto::getEmail, EMAIL_PATTERN, /*EMAIL_ERROR_MESSAGE, */entity, new RegisterPageEmailErrorException(EMAIL_ERROR_MESSAGE));
+//        validateByParam(UserDto::getFirstName, FIRST_NAME_PATTERN, FIRST_NAME_ERROR_MESSAGE, entity);
+//        validateByParam(UserDto::getLastName, LAST_NAME_PATTERN, LAST_NAME_ERROR_MESSAGE, entity);
+//        validateByParam(UserDto::getPassword, PASSWORD_PATTERN, PASSWORD_ERROR_MESSAGE, entity);
+//        validateByParam(UserDto::getConfirmationPassword, PASSWORD_PATTERN, CONFIRMATION_PASSWORD_ERROR_MESSAGE, entity);
+
+
+        validateByParam(UserDto::getFirstName, FIRST_NAME_PATTERN, entity, new RegisterPageFirstNameErrorException(FIRST_NAME_ERROR_MESSAGE));
```
Comment (MaxWel2018) [inline src/main/java/com/vladnickgofj/hotel/validator/UserValidator.java:33]: why did you create new excepting for each field?

**Review by 1SpringIsComing1 — APPROVED** [review]

**Review by MaxWel2018 — APPROVED** [review]

**Pull request merged**